src/components/organisms/multiStepForm/MultiStepFormNEW.tsx

```diff
@@ -0,0 +1,107 @@
+"use client";
+
+import React, { ReactElement, useState } from "react";
+import { Button } from "~/components/atoms/Button";
+import { useForm, SubmitHandler } from "react-hook-form";
+import FormComponent from "../form/formComponent/FormComponent";
+
+export interface FormStep {
+  name: string;
+  form: ReactElement;
+}
+
+interface FormProps {
+  forms: FormStep[];
+}
+type Inputs = {
+  username: string;
+  password: string;
+  password2: string;
+  email: string;
+  phone: string;
+  creditCard: string;
+  billingAddr: string;
+};
+const MultiStepForm: React.FC<FormProps> = ({ forms }) => {
+  const [currentStep, setCurrentStep] = useState(0);
+  let isLastStep = false;
+  let newStep = currentStep;
+
+  const methods = useForm<Inputs>({
+    //shouldUseNativeValidation: true,
+    // reValidateMode: "onSubmit",
+    mode: "all",
+  });
+  console.log(newStep);
+  const numSteps = forms.length;
+
+  const nextStep = () => {
+    if (currentStep < numSteps - 1) {
+      newStep += 1;
+
+      setCurrentStep(currentStep + 1);
+    } else isLastStep = true;
+  };
+
+  const prevStep = () => {
+    if (currentStep > 0) {
+      newStep -= 1;
+      setCurrentStep(currentStep - 1);
+    }
+  };
+
+  const onSubmit: SubmitHandler<Inputs> = (data) => {
+    //nextStep();
+    //console.log(typeof data);
+    if (currentStep == numSteps - 1 && isLastStep) alert(JSON.stringify(data));
+    // Add submission logic here
```

Comment:
This logic must be provided from parent component. Because handle submit usually is calling some API. I can see two possible solutions.
1. Add another prop to `MultiStepForm`. Besides `form`, add `formSubmission` void function which is going to handle the submission process.
2. Change the `FormStep`.
   1. Rename it to `FormAction` 
   2. It should contain array of `FormSteps` and `onSubmit` function
 
```typescript
export type FormAction {
  formSteps: FormStep[];
  onSubmit: void();
};
 
export type FormStep {
  name: string;
  form: ReactElement;
}
```